lib/smooth/core/io/spi/Master.cpp

```diff
@@ -23,63 +23,101 @@ namespace smooth::core::io::spi
+                            int transfer_size,
+                            gpio_num_t quadwp_io_num,
+                            gpio_num_t quadhd_io_num)
     {
+        std::lock_guard<std::mutex> lock(guard);
+
+        spi_host = host;
+        dma_channel = dma_chl;
+
         bus_config.mosi_io_num = mosi;
         bus_config.miso_io_num = miso;
         bus_config.sclk_io_num = clock;
         bus_config.quadwp_io_num = quadwp_io_num;
         bus_config.quadhd_io_num = quadhd_io_num;
         bus_config.max_transfer_sz = transfer_size;
-    }
 
-    Master::~Master()
-    {
-        deinitialize();
+        spi_bus_initialized = false;
+
+        if (spi_host == VSPI_HOST)
+        {
+            spi_bus_initialized = do_vspi_initialization();
+        }
+
+        if (spi_host == HSPI_HOST)
+        {
+            spi_bus_initialized = do_hspi_initialization();
+        }
+
+        return spi_bus_initialized;
     }
 
-    bool Master::initialize()
+    bool Master::do_vspi_initialization()
     {
-        std::lock_guard<std::mutex> lock(guard);
-        do_initialization();
+        if (!vspi_initialized)
+        {
+            vspi_initialized = spi_bus_initialize(spi_host, &bus_config, dma_channel) == ESP_OK;
 
-        return initialized;
+            if (!vspi_initialized)
+            {
+                Log::error(log_tag, "VSPI_HOST Initialization failed");
+            }
+            else
+            {
+                Log::info(log_tag, "VSPI initialized, Host {}, DMA {}", spi_host, dma_channel);
+            }
+        }
+
+        return vspi_initialized;
     }
 
-    void Master::do_initialization()
+    bool Master::do_hspi_initialization()
     {
-        if (!initialized)
+        if (!hspi_initialized)
         {
-            initialized = spi_bus_initialize(host, &bus_config, dma_channel) == ESP_OK;
+            hspi_initialized = spi_bus_initialize(spi_host, &bus_config, dma_channel) == ESP_OK;
 
-            if (!initialized)
+            if (!hspi_initialized)
             {
-                Log::error(log_tag, "Initialization failed");
+                Log::error(log_tag, "HSPI_HOST Initialization failed");
             }
             else
             {
-                Log::verbose(log_tag, "SPI initialized, Host {}, DMA {}", host, dma_channel);
+                Log::info(log_tag, "HSPI initialized, Host {}, DMA {}", spi_host, dma_channel);
             }
         }
+
+        return hspi_initialized;
     }
 
-    void Master::deinitialize()
+    void Master::deinitialize(spi_host_device_t spi_host)
     {
-        if (initialized)
+        if ((spi_host == VSPI_HOST) & (vspi_initialized))
+        {
+            vspi_initialized = false;
+            spi_bus_free(VSPI_HOST);
+        }
+
+        if ((spi_host == HSPI_HOST) & (hspi_initialized))
         {
-            initialized = false;
-            spi_bus_free(host);
+            hspi_initialized = false;
+            spi_bus_free(HSPI_HOST);
```

Comment:
With the previous comment in mind, this can be written as

if (initialized)
        {
            initialized = false;
            spi_bus_free(spi_host);
        }

Reply:
See previous comment